src/features/cv/cvSlice.ts

```diff
@@ -16,9 +18,36 @@ const cvSlice = createSlice({
     setAddress(state, action: PayloadAction<string>) {
       state.personalInfo.address = action.payload;
     },
-    setIntroResultFromAI(state, action: PayloadAction<string>) {
+    setDescription(state, action: PayloadAction<string>) {
       state.personalInfo.description = action.payload;
     },
+    addWorkingExperience(state, action: PayloadAction<ArrayElementType<WorkExperienceType>>) {
+      state.workExperience = [...(state.workExperience || []), action.payload];
+    },
+    setCompanyName(state, action: PayloadAction<{ id: string | number; companyName: string }>) {
+      state.workExperience = state?.workExperience?.map((company) => {
+        if (company.id === action.payload.id) {
```

Comment:
could have only one reducer for every workExpirience field
```
return company.id === action.payload.id ? { ...company, ...action.payload } : company;
```

Reply:
Agree, and the same we can make for personalInfo, education, and other group states that we have which represent something common